core/dbt/adapters/factory.py

```diff
@@ -89,7 +90,11 @@ def load_plugin(self, name: str) -> Type[Credentials]:
     def register_adapter(self, config: AdapterRequiredConfig) -> None:
         adapter_name = config.credentials.type
         adapter_type = self.get_adapter_class_by_name(adapter_name)
-
+        adapter_version = import_module(f".{adapter_name}.__version__", "dbt.adapters").version
+        adapter_version_specifier = VersionSpecifier.from_version_string(adapter_version).to_version_string()
```

Comment:
non-blocking suggestion: Assuming there are no circular dependencies, what do you think about passing the 'raw' `adapter_version` string directly to the `AdapterRegistered` event and using `VersionSpecifier` within the event to format the string in the message? That way formatting concerns are handled closer to the message relying on them.

Reply:
I agree it would be cleaner but looks like it does introduce a circular dependency

Reply:
dang!